.github/workflows/on-push-do-docs.yml

```
@@ -0,0 +1,99 @@
+name: on-push-do-docs
+
+on:
+  push:
+    branches: [main]
+    paths: [ "src/Snippets/**" ]
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  update-docs:
+    name: update-docs
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: write
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
+
+      - name: Setup .NET SDK
+        uses: actions/setup-dotnet@3447fd6a9f9e57506b15f895c5b76d3b197dc7c2 # v3.2.0
+
+      - name: Update documentation
+        id: update-docs
+        shell: pwsh
+        env:
+          DOTNET_CLI_TELEMETRY_OPTOUT: true
+          DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true        
+        run: |
+          $ErrorActionPreference = "Stop"
+          $ProgressPreference = "SilentlyContinue"
+  
+          dotnet tool restore
+          dotnet mdsnippets
+  
+          $GitStatus = (git status --porcelain)
+          if ([string]::IsNullOrEmpty($GitStatus)) {
+            Write-Host "No changes to commit."
+            exit 0
+          }
+
+          $TimeStamp = Get-Date -Format "yyyy-MM-dd-HH-mm"
+          $BranchName = "docs/update-docs-$TimeStamp"
+          echo "branchName=$BranchName" >> $env:GITHUB_OUTPUT
```

If the last line of the step doesn't need the `echo` then we can remove it from here too.

```suggestion
          "branchName=$BranchName" >> $env:GITHUB_OUTPUT
```